File: 5/05426.py
from math import sqrt

for i in range(int(input())):
    s = input()
    l = len(s); size = int(sqrt(l))

    index = 0; letter = []
    for j in range(size):
        letter.append(s[index:index+size])
        index += size

    for j in range(size-1, -1, -1):
        for k in range(size):
            print(letter[k][j], end='')
    print()

File: 5/05426.cpp
#include <iostream>
#include <cmath>
using namespace std;

char letter[100][100];

int main() {
	int T; cin >> T;

	for (int i = 0; i < T; i++) {
		string s; cin >> s;
		int len = s.length();
		int size = sqrt(len);

		int index = 0;
		for (int j = 0; j < size; j++)
			for (int k = 0; k < size; k++)
				letter[j][k] = s[index++];

		for (int j = size - 1; j >= 0; j--)
			for (int k = 0; k < size; k++)
				cout << letter[k][j];
		cout << endl;
	}
}
